debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/tools/ddrinteractive/RomClassWalker.java

```diff
@@ -1054,10 +1055,10 @@ long allSlotsInMethodDebugInfoDo(U32Pointer cursor) throws CorruptDataException
 				LocalVariableTable values = variableInfoValuesIterator.next();
 
 				// Need to walk the name and signature to add them to the UTF8 section
-				classWalkerCallback.addSlot(clazz, SlotType.J9_UTF8, values.getName(), "name");
-				classWalkerCallback.addSlot(clazz, SlotType.J9_UTF8, values.getSignature(), "getSignature");
+				classWalkerCallback.addSlot(clazz, SlotType.J9_ROM_UTF8, values.getNameSrp(), "name");
+				classWalkerCallback.addSlot(clazz, SlotType.J9_ROM_UTF8, values.getSignatureSrp(), "signature");
 				if (values.getGenericSignature().notNull()) {
-					classWalkerCallback.addSlot(clazz, SlotType.J9_UTF8, values.getGenericSignature(), "getGenericSignature");
+					classWalkerCallback.addSlot(clazz, SlotType.J9_ROM_UTF8, values.getGenericSignatureSrp(), "genericSignature");
```

Comment:
I think these should continue to use `J9_UTF8` to benefit from the extra sanity checking in [LinearDumper.java](https://github.com/eclipse/openj9/blob/master/debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/tools/ddrinteractive/LinearDumper.java#L281).
The corresponding arguments (`getNameSrp()`, etc. are correct).

Reply:
As far as I understand the code in `LinearDumper.java`, the strings pointed to by `J9_ROM_UTF8` slots go through the same bound checking as with `J9_UTF8`. The difference between them is that `J9_ROM_UTF8` also handles the SRP, not just the string it points to, so the resulting tree of regions will actually be more complete. The rest of the code uses `J9_ROM_UTF8`, and it would actually be possible to remove `J9_UTF8` altogether after this change. I can only see `J9_UTF8` being necessary when a `J9UTF8` string is stored "inline" somewhere with no SRP pointing to it, but there are no such cases right now as far as I can tell.

Reply:
I looked at the code I referenced in `LinearDumper` again: I think you're right.